src/host/tcp.h

```diff
@@ -327,14 +360,37 @@ namespace asynchost
     bool init()
     {
       assert_status(FRESH, FRESH);
-      int rc;
 
+      int rc;
       if ((rc = uv_tcp_init(uv_default_loop(), &uv_handle)) < 0)
       {
         LOG_FAIL_FMT("uv_tcp_init failed: {}", uv_strerror(rc));
         return false;
       }
 
+      if (is_client)
+      {
+        uv_os_sock_t sock;
+        if ((sock = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP)) == -1)
+        {
+          LOG_FAIL_FMT("socket creation failed: {}", strerror(errno));
```

Comment:
If we have the address easily available, it'd be really good to add it to this log line.

Reply:
It's too early at this point as (the address is specified later, on connect.